### PR TITLE
Add a BaseError parent class to errors

### LINKED ISSUE #46: Introduce a CamtParser::Errors base class
Hi! 

I was looking at the collection of errors [the Errors module](https://github.com/viafintech/camt_parser/blob/master/lib/camt_parser/errors.rb).

All of them inherit directly from Ruby's StandardError. 

What if they inherited from a `class CamtParserError < StandardError` exception class? That way, I'd be able to catch that one, in my application code.

### PR DESCRIPTION
**Pull request by olleolleolle**
Fixes #46.

### CODE REVIEW

**Comment by olleolleolle** [comment]
"bundler-1.17.3/lib" - seems like we need to allow Bundler 2 somewhere.

**Comment by tobischo** [comment]
Or we need to install bundler 1.17.3 for the CI jobs

**Comment by olleolleolle** [comment]
I'll make a Gemfile.lock update in a commit here.

**Comment by tobischo** [comment]
The problem is that we are not using bundler 2 (yet).

Therefore just updating it from bundler 1 to bundler 2 will break it for us

**Comment by olleolleolle** [comment]
@tobischo Well, the real problem wasn't the Bundler version, though, it was that the Gemfile.lock had not been fully updated, so that during the installation in CI, the Gemfile.lock changed (and that fails when --deployment flag is enabled, which the ruby/setup-ruby is enabling).

**Comment by tobischo** [comment]
Yeah, that was only started recently that it does not support bundler 1 anymore

I can look into it later

**Comment by olleolleolle** [comment]
@tobischo No, all is well, this is ready to go!

File: Gemfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    camt_parser (2.11.0)
+    camt_parser (2.11.1)
```
Comment (olleolleolle) [inline Gemfile.lock:4]: This was the diff which stopped the `bundle install` in CI. Now, we'll have no issues.

Comment (tobischo) [reply Gemfile.lock:4]: Ah, damn, I broke that the other day then 😄 
Thank you!

**Comment by olleolleolle** [comment]
@tobischo Thanks!

**Pull request merged**